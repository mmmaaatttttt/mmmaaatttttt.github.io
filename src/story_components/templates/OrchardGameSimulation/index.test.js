import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { OrchardGameSimulation } from ".";

describe("smoke and snapshot tests", () => {
  it("renders successfully", () => {
    render(<OrchardGameSimulation />);
  });

  it("renders first demonstration successfully", () => {
    const { asFragment } = render(<OrchardGameSimulation />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("state changes", () => {
  it("manages button state correctly", () => {
    const { debug, queryByText } = render(<OrchardGameSimulation />);
    const playBtn = queryByText("Play");

    const resetBtn = queryByText("Reset Simulation");

    // expect both buttons to appear initially
    expect(playBtn).toBeInTheDocument();
    expect(resetBtn).toBeInTheDocument();

    fireEvent.click(playBtn);

    expect(playBtn).toHaveTextContent("Pause");
    expect(resetBtn).not.toBeInTheDocument();

    fireEvent.click(playBtn);

    expect(playBtn).toHaveTextContent("Play");
    expect(resetBtn).toBeInTheDocument();
    expect(wrapper.state().playing).toBe(false);
  });
});
