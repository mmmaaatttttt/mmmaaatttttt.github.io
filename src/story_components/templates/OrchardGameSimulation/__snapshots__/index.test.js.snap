// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests renders first demonstration successfully 1`] = `
<styled.div
  fullWidthAt="small"
  margin="0 auto"
  width="80%"
>
  <styled.div
    column={false}
    cross="stretch"
    flex={1}
    main="center"
    margin="0"
    shouldWrap={false}
    textAlign="left"
    width="auto"
  >
    <Button
      color="white"
      disabled={false}
      large={false}
      onClick={[Function]}
    >
      Play
    </Button>
    <Button
      color="#FF3C23"
      disabled={false}
      large={false}
      onClick={[Function]}
    >
      Reset Simulation
    </Button>
  </styled.div>
  <HorizontalBar
    data={
      Array [
        Object {
          "color": "#52A081",
          "size": 0,
          "tooltipText": "Games Won: 0",
        },
        Object {
          "color": "#FF3C23",
          "size": 0,
          "tooltipText": "Games Played: 0",
        },
        Object {
          "color": "#BBBBBB",
          "size": 0,
          "tooltipText": "Win Percentage: 0.0%",
        },
      ]
    }
    key="0"
    title="Most Plentiful Strategy: 0.0%"
  />
  <HorizontalBar
    data={
      Array [
        Object {
          "color": "#52A081",
          "size": 0,
          "tooltipText": "Games Won: 0",
        },
        Object {
          "color": "#FF3C23",
          "size": 0,
          "tooltipText": "Games Played: 0",
        },
        Object {
          "color": "#BBBBBB",
          "size": 0,
          "tooltipText": "Win Percentage: 0.0%",
        },
      ]
    }
    key="1"
    title="Least Plentiful Strategy: 0.0%"
  />
  <HorizontalBar
    data={
      Array [
        Object {
          "color": "#52A081",
          "size": 0,
          "tooltipText": "Games Won: 0",
        },
        Object {
          "color": "#FF3C23",
          "size": 0,
          "tooltipText": "Games Played: 0",
        },
        Object {
          "color": "#BBBBBB",
          "size": 0,
          "tooltipText": "Win Percentage: 0.0%",
        },
      ]
    }
    key="2"
    title="Random Strategy: 0.0%"
  />
  <HorizontalBar
    data={
      Array [
        Object {
          "color": "#52A081",
          "size": 0,
          "tooltipText": "Games Won: 0",
        },
        Object {
          "color": "#FF3C23",
          "size": 0,
          "tooltipText": "Games Played: 0",
        },
        Object {
          "color": "#BBBBBB",
          "size": 0,
          "tooltipText": "Win Percentage: 0.0%",
        },
      ]
    }
    key="3"
    title="Favorite Color Strategy: 0.0%"
  />
</styled.div>
`;
